jwst/coron/median_replace_img.py

```diff
@@ -89,9 +91,15 @@ def median_replace_img(img_model, box_size, bad_bitvalue):
         img_int = img_model.data[nimage]
         img_dq = img_model.dq[nimage]
         bad_locations = np.where(np.bitwise_and(img_dq, bad_bitvalue))
-
+        # if it's MIRI, only use the bad locations that are in the science aperture
+        # set the others to 0 with no logging message
+        # bad_locations, zero_locations = filter_bad_locations(img_model, bad_locations)
+        bad_locations, non_science = separate_non_science_pixels(img_dq, bad_locations)
+        img_model.data[nimage, non_science[0], non_science[1]] = 0
```

Comment:
I'm confused by the fact that values in `img_model.data` are being set to zero here, but later in line 105 it's `img_int` that's passed to the `median_fill_value` function, which has not had any values reset. Is that correct? Or does it work out properly because `img_int` is just a view of (pointer to) `img_model.data` and hence it has the same values?

Reply:
Ah, that's a good point. It works because it's all passed by reference, but I changed img_model.data to img_int for consistancy.